clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2276,6 +2337,141 @@ private Node leader(PartitionInfo partitionInfo) {
+        TopicCollection topics,
+        DescribeTopicsOptions options,
+        AdminResultsSubscriber<DescribeTopicPartitionsResult> subscriber) {
+        if (topics instanceof TopicIdCollection) {
+            subscriber.onError(
+                new IllegalArgumentException("Currently the describeTopics subscription mode does not support topic IDs.")
+            );
+            return;
+        }
+        if (!(topics instanceof TopicNameCollection)) {
+            subscriber.onError(
+                new IllegalArgumentException("The TopicCollection: " + topics + " provided did not match any supported classes for describeTopics.")
+            );
+            return;
+        }
+
+        TreeSet<String> topicNames = new TreeSet<>();
+        ((TopicNameCollection) topics).topicNames().forEach(topicName -> {
+            if (topicNameIsUnrepresentable(topicName)) {
+                subscriber.onNext(DescribeTopicPartitionsResult.onError(topicName, new InvalidTopicException("The given topic name '" +
+                    topicName + "' cannot be represented in a request.")));
+            } else {
+                topicNames.add(topicName);
+            }
+        });
+
+        RecurringCall call = new RecurringCall(
+            "DescribeTopics-Recurring",
+            calcDeadlineMs(time.milliseconds(), options.timeoutMs()),
+            runnable
+        ) {
+
+            Map<String, TopicRequest> pendingTopics = new TreeMap<>();
+            Iterator<String> pendingTopicIterator = topicNames.iterator();
+
+            String partiallyFinishedTopicName = "";
+            int partiallyFinishedTopicNextPartitionId = -1;
+
+            @Override
+            Call generateCall() {
+                return new Call("describeTopics", this.deadlineMs, new LeastLoadedNodeProvider()) {
+                    @Override
+                    DescribeTopicPartitionsRequest.Builder createRequest(int timeoutMs) {
+                        DescribeTopicPartitionsRequestData request = new DescribeTopicPartitionsRequestData()
+                            .setResponsePartitionLimit(options.partitionSizeLimitPerResponse());
+
+                        if (!partiallyFinishedTopicName.isEmpty()) {
+                            request.setCursor(new DescribeTopicPartitionsRequestData.Cursor()
+                                .setTopicName(partiallyFinishedTopicName)
+                                .setPartitionIndex(partiallyFinishedTopicNextPartitionId)
+                            );
+                        }
+
+                        for (int ii = pendingTopics.size(); ii < options.partitionSizeLimitPerResponse() && pendingTopicIterator.hasNext(); ++ii) {
+                            String topicName = pendingTopicIterator.next();
+                            pendingTopics.put(topicName, new TopicRequest().setName(topicName));
+                        }
+                        request.setTopics(pendingTopics.values().stream().collect(Collectors.toList()));
```

Comment:
This is a hint more concise, but arguably just preference:

```suggestion
                        request.setTopics(new ArrayList<>pendingTopics.values());
```